### PR TITLE
[DOC] Document `Marshal#load` parameter `freeze:`

### PR DESCRIPTION
**Pull request by zverok**
Screenshot:
image

### CODE REVIEW

**Comment by zverok** [comment]
@byroot Please review

File: marshal.rb [hunk 1]
```diff
@@ -11,6 +11,25 @@ module Marshal
   #
   # Never pass untrusted data (including user supplied input) to this method.
   # Please see the overview for further details.
+  #
+  # If the <tt>freeze: true</tt> argument is passed, deserialized object would
+  # be deeply frozen. Note that it will lead to more efficient memory usage due to
```
Comment (byroot) [inline marshal.rb:16]:
```suggestion
  # be deeply frozen. Note that it may lead to more efficient memory usage due to
```
?

**Review by byroot — APPROVED** [review]

File: marshal.rb [hunk 2]
```diff
@@ -11,6 +11,25 @@ module Marshal
...
+  #    deserialized.map(&:frozen?)
+  #    # => [false, false, false, false]
+  #    deserialized.map(&:object_id)
+  #    # => [1023900, 1023920, 1023940, 1023960] -- 4 different objects
+  #
+  #    deserialized = Marshal.load(serialized, freeze: true)
+  #    deserialized.map(&:frozen?)
+  #    # => [true, true, true, true]
+  #    deserialized.map(&:object_id)
+  #    # => [1039360, 1039380, 1039360, 1039380] -- only 2 different objects, object_ids repeating
```
Comment (nobu) [inline marshal.rb:31]: These lines seem indented more than above `call-seq`, intentional?

Comment (zverok) [reply marshal.rb:31]: Ugh, no. Updated

**Pull request merged**